tutorial/guias/Paso_a_paso_DIA_6.md

```diff
@@ -0,0 +1,375 @@
+
+    Returns:
+        bool: True in case the snake's head is touching the boundaries, False otherwise.
+    """
+    return (
+        self.snake.position[0] < 0
+        or self.snake.position[0] > self.window_x - 10
+        or self.snake.position[1] < 0
+        or self.snake.position[1] > self.window_y - 10
+    )
+```
+
+Esta función no recibe parámetros y devuelve un valor booleano (`True` - verdadero -  o `False` - falso). La serpiente toca un borde cuando la posición horizontal o vertical de su cabeza está en uno de esos bordes. En nuestro caso, la posición de la cabeza se almacena en el campo `position` del objeto `snake`. Recordemos que este campo es una tupla y que la posición horizontal está en la posición 0 de la tupla mientras que la posición vertical está en la posición 1.
+
+El borde izquierdo de la pantalla se identifica con el número 0 y el borde derecho estará justo 10 píxeles antes del valor del ancho de la pantalla. De la misma forma, el borde inferior de la pantalla se identifica con 0 y el superior justo 10 píxeles antes del valor de la altura de la pantalla. La serpiente, por tanto, tocará un borde cuando su posición horizontal o vertical sea menor que 0 o mayor que los valores para los bordes derecho y superior respectivamente. Esto lo expresamos en la función con cuatro condiciones unidas por el operador `or` que indica que toda la expresión será verdadera si al menos una de las condiciones es verdadera.
+
+>**Nota :pen:** Luego de la palabra reservada **return** hemos usado paréntesis para poder escribir una condición por línea. De esta forma no tenemos que escribir una línea muy extensa y nuestro código se puede leer más fácilmente.
+
+La siguiente función es `_touch_body`, que verifica si la serpiente ha tocado su propio cuerpo:
+
+```python
+def _touch_body(self) -> bool:
+    """Checks wheter the snake's head is touching its body.
+
+    Returns:
+        bool: True in case the snake's head is touching its body, False otherwise.
+    """
+    return any(self.snake.position == block for block in self.snake.body[1:])
+```
+
+Hay varios detalles en esta función. Analicemos primero lo que tenemos dentro de los paréntesis en la sentencia **return**:
+
+```python
+self.snake.position == block for block in self.snake.body[1:]
+```
+
+Como ya sabemos, esto es una forma de crear una colección ordenada de elementos. En este caso, cada elemento será `True` o `False`. Será verdadero (`True`) si la posición de la cabeza es igual a la tupla `block` (ya veremos de dónde sale `block`), y falso (`False`) en otro caso. La tupla `block` no es más que cada elemento del cuerpo de la serpiente (campo `body` del objeto `snake`). No obstante, recordemos que el primer elemento del cuerpo de la serpiente es la cabeza pero no nos interesa comprobar que la cabeza haya chocado consigo misma, por tanto solamente consideramos los elementos de esta lista a partir del segundo. Esto lo logramos con `self.snake.body[1:]`.
```

Comment:
Maybe we should add a small reference to what is `slicing` in Python

Reply:
It could also be a new blog posts hehe